### PR TITLE
Skip scheduling recent failed tasks if an earlier task is already InProgress

### PR DESCRIPTION
**Pull request by keyonghan**
Fixes: https://github.com/flutter/flutter/issues/110716

### CODE REVIEW

**Review by ricardoamador — APPROVED** [review]
LGTM

File: app_dart/lib/src/service/scheduler/policy.dart [hunk 1]
```diff
@@ -56,6 +56,11 @@ class BatchPolicy implements SchedulerPolicy {
     required DatastoreService datastore,
   }) async {
     final List<Task> recentTasks = await datastore.queryRecentTasksByName(name: task.name!).toList();
+    // Skip scheduling if there is already a running task.
+    if (recentTasks.any((Task task) => task.status == Task.statusInProgress)) {
```
Comment (ricardoamador) [inline app_dart/lib/src/service/scheduler/policy.dart:60]: Not a code review comment but is it possible that a running task can still be active outside the "last 6 records" fetched? Meaning a task is running but the most recent 6 show a status other than Task.statusInProgress?

Comment (keyonghan) [reply app_dart/lib/src/service/scheduler/policy.dart:60]: That's a very good point, and the answer is yes.
We will skip scheduling that case as well. At any specific time, we want only one task is running (In Progress).
If we have an old task running (beyond the 6), we will skip it in the webhook scheduling. When it finishes, we rely on the back filler to pick up the most recent New task to run.

Comment (ricardoamador) [reply app_dart/lib/src/service/scheduler/policy.dart:60]: Ah okay cool! Thanks for the answer!

**Pull request merged**